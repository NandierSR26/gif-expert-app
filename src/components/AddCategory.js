import React, { useState } from 'react';
import PropTypes from 'prop-types';

/* componente con el formulario y funcionalidad para aÃ±adir 
una nueva categoria de gifs*/
export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    // extrae lo que el usuario escriba en el input
    const handleInputChange = (e) => {
        // console.log(e.target.value);
        setInputValue(e.target.value)
    }

    // realiza una busqueda y actualiza el state con los resultados
    const handleSubmit = e => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories(cat => [inputValue, ...cat]);
            setInputValue('')
        }

    }
    return (
        <>
            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    value={inputValue}
                    onChange={handleInputChange}
                />
            </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
