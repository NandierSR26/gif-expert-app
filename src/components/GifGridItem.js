import React from 'react';

/* componente de trajeta de cada gif, se llama desde GifGrid
y se itera con todas las imagenes, ademas se envian el 
id, titulo y urlcomo props */
export const GifGridItem = ( {id, title, url} ) => {

    // console.log(id, title, url);

    return (
        <div className="card animate__animated animate__fadeInLeft">
            <img
                src={url}
                alt={title}
            />
            <div className="card-text">
                <p>{title}</p>
            </div>
        </div>
    )
}
