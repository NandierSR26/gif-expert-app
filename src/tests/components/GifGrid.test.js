import React from 'react';
import { GifGrid } from "../../components/GifGrid";
import { shallow } from "enzyme";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('pruebas en GifGrid', () => {

    const category = 'naruto'
    test('debe mostrase correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category }/> )
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier Cosa'
        },
        {
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier Cosa'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category={ category }/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
        
    });
    
    
});
