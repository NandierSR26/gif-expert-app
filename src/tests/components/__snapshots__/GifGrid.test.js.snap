// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGrid debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    naruto
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`pruebas en GifGrid debe mostrase correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    naruto
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
